UPGRADING.md

```diff
@@ -0,0 +1,119 @@
+  api_url      = "https://api.fr.cloud.gov"
+  user         = var.cf_user
+  password     = var.cf_password
+}
+
+provider "cloudfoundry-community" {
+  api_url      = "https://api.fr.cloud.gov"
+  user         = var.cf_user
+  password     = var.cf_password
+}
+```
+
+The v1 modules should properly select the `cloudfoundry-community` provider, but if they don't you may need to [explicitely set the provider](https://developer.hashicorp.com/terraform/language/modules/develop/providers#passing-providers-explicitly):
+
+```
+module "database" {
+  source = "github.com/gsa-tts/terraform-cloudgov//database?ref=v1.1.0"
+  providers = {
+    cloudfoundry = cloudfoundry-community
+  }
+
+  # ...
+}
+```
+
+## Provider Upgrades
+
+Follow the steps in the [cloudfoundry provider migration guide](https://github.com/cloudfoundry/terraform-provider-cloudfoundry/blob/main/migration-guide/Readme.md) to migrate an existing use of the v1 module to v2. As an example, here are the steps for upgrading a database module:
+
+1. Update source line to point to v2 module and change `cf_space_name` to `cf_space_id`
+1. Verify that `terraform validate` passes
+1. Run: `terraform state show module.database.cloudfoundry_service_instance.rds | grep -m 1 id` and copy the ID
+1. Run: `terraform state rm module.database.cloudfoundry_service_instance.rds`
+1. Run: `terraform import module.database.cloudfoundry_service_instance.rds ID_FROM_STEP_3`
+1. Run: `terraform apply` to fill in new computed attributes
+
+## Module Changes
+
+### Common Changes
+
+1. Check the `variables.tf` and `outputs.tf` files for each module for new names of variables and outputs. There should not be any variables or outputs that kept the same name but changed behavior.
+
+### Egress Proxy
+
+Egress Proxy no longer sets up network policies between the proxy and client apps, and does not create a User Provided Service Instance to deliver the credentials to the app. It is the developer's responsibility to do those things in the root module to better handle circular dependencies between creating the client app(s) and the proxy.
+
+To setup the network policy in your root module, add:
+
+```
+resource "cloudfoundry_network_policy" "egress_policy" {
+  provider = cloudfoundry-community
+  policy {
+    source_app      = cloudfoundry_app.client_app.id # assumes you're deploying the client app with terraform
+    destination_app = module.egress_proxy.app_id
+    port            = module.egress_proxy.https_port
+  }
+}
+```
+
+To add a UPSI:
+
+```
+resource "cloudfoundry_service_instance" "egress_proxy_credentials" {
+  name        = "egress-proxy-credentials"
+  space       = module.app_space.space_id
+  type        = "user-provided"
+  credentials = module.egress_proxy.json_credentials
+}
+```
+
+### Clamav
+
+Clamav no longer sets up network policies between the clamav app and client apps. It is the developer's responsibility to set this up to better handle circular dependencies between the various apps.
+
```

Comment:
We should probably duplicate [these lines](https://github.com/GSA-TTS/terraform-cloudgov/pull/57/files/169e6ed7c05baff02d750cad1e54428568d190c9..35a63aa2d5c4d2e6d1c8ddee9f75e250813958ef#diff-f411a5009131db5a6c7242b49ca78488f2d274ef9e5e5f4de9262a42108a5386R77) here in case the reader is not using the egress proxy and skipped past the section above.